usaspending_api/download/v2/download_admin.py

```diff
@@ -25,12 +26,19 @@ def get_download_job(self, queryset_filter):
         self.download_job = query_database_for_record(queryset_filter)
 
     def restart_download_operation(self):
+        self.update_download_job(
+            error_message=None,
+            file_size=0,
+            job_status_id=JOB_STATUS_DICT["queued"] if process_is_local() else JOB_STATUS_DICT["ready"],
+            number_of_columns=0,
+            number_of_rows=0,
+            update_date=datetime.now(timezone.utc),
+        )
+
         if process_is_local():
-            self.update_download_job(job_status_id=JOB_STATUS_DICT["ready"], error_message=None)
             download_generation.generate_download(download_job=self.download_job)
         else:
             self.push_job_to_queue()
```

Comment:
Does it matter that you are now updating the download job before it gets pushed to the queue?

Reply:
Yeah, if this status was currently used, or if adding things to the queue was prone to errors.

I have seen evidence of neither.